geode-core/src/integrationTest/java/org/apache/geode/internal/net/SSLSocketIntegrationTest.java

```diff
@@ -187,6 +325,33 @@ public void handshakeCanTimeoutOnServer() throws Throwable {
     throw serverException;
   }
 
+  @Test(expected = SocketTimeoutException.class)
+  public void handshakeWithPeerCanTimeout() throws Throwable {
+    ServerSocketChannel serverChannel = ServerSocketChannel.open();
+    serverSocket = serverChannel.socket();
+
+    InetSocketAddress addr = new InetSocketAddress(localHost, 0);
+    serverSocket.bind(addr, 10);
+    int serverPort = this.serverSocket.getLocalPort();
+
+    this.serverThread = startServerNIO(this.serverSocket, 1000);
+
+    Socket socket = new Socket();
+    await().atMost(10, TimeUnit.SECONDS).until(() -> {
```

Comment:
I'm not sure you should use 10 SECONDS. GeodeAwaitility defaults to 5 MINUTES which is larger than it needs to be just to avoid false failures on amazingly slow machines or long GC pauses in the middle of a test.